packages/react-router-dom/index.tsx

```diff
@@ -1216,6 +1241,49 @@ function useDataRouterState(hookName: DataRouterStateHook) {
   return state;
 }
 
+function useFetcherDataLayer() {
+  let fetcherRefs = React.useRef<Map<string, number>>(new Map());
+  let fetcherData = React.useRef<Map<string, any>>(new Map());
```

Comment:
I _think_ we're OK using a ref here but I need to bang on some edge cases to be sure - all tests and such pass as-is which is reassuring.  The idea is that anytime a `fetcher.data` needs to update - a fetcher will have changed `state` which will be what triggers the re-render, and we don't need ` useState` version of `fetcherData`.